spec/driverProviderTest.js

```diff
@@ -0,0 +1,138 @@
+/**
+ * Sanity integration tests for Driver Providers.
+ *
+ * Assumed setup:
+ * - selenium server running locally at http://localhost:4444
+ * - selenium jar and chromedriver in protractor/selenium, where
+ *   webdriver-manager stores them.
+ * - if you want to test saucelabs, test with --sauceUser and --sauceKey
+ * - if you want to test browserstack driverProvider, test with 
+     --browserstackUser and --browserstackKey
+ * You should verify that there are no lingering processes when these tests
+ * complete.
+ */
+
+const argv = require('optimist').argv;
+const env = require('./environment');
+
+const Direct = require('../built/driverProviders/direct').Direct;
+const Hosted = require('../built/driverProviders/hosted').Hosted;
+const Local = require('../built/driverProviders/local').Local;
+const Sauce = require('../built/driverProviders/sauce').Sauce;
+const BrowserStack = require('../built/driverProviders/browserStack').BrowserStack;
+
+const testDriverProvider = async (driverProvider) => {
+  await driverProvider.setupEnv();
+  const driver = driverProvider.getNewDriver();
+  await driver.get('about:blank');
+  const url = await driver.getCurrentUrl();
+  if (url != 'about:blank') {
+    throw new Error(`url was not about:blank, instead found ${url}`);
+  }
+  
+  if (driverProvider.updateJob) {
+    await driverProvider.updateJob({'passed': true});
+    await driverProvider.teardownEnv();
+  } else {
+    await driverProvider.teardownEnv();
+  }
+};
+
+const chromeConfig = {
+  capabilities: {
+    browserName: 'chrome'
+  }
+};
+
+testDriverProvider(new Direct(chromeConfig)).
+    then(() => {
+      console.log('direct.dp with chrome working!');
+    }, (err) => {
+      console.log('direct.dp with chrome failed with', err);
```

Comment:
Should these console.error, or exit with non-zero?

Reply:
These should fail. I'll throw errors here.